Set docs check job status to fail when htmltest fails

DamianMaslanka5 opened the pull request:
I fixed reporting status of the docs check job, also I fixed reported issues related to `stochasticlogisticregression`.

But html contains `null` char (also in links), pages look ok, but htmltest stopped working

![image](https://github.com/user-attachments/assets/ff9556a8-14ce-441b-a8a6-0c79a896ea9e)

This issue might be fixed in docusaurus 3.7 (and react 19?), see https://github.com/facebook/docusaurus/issues/9985

Logs from master https://github.com/ClickHouse/ClickHouse/actions/runs/12768456850/job/35596857109:

```
htmltest started at 12:56:19 on /output_path/build
========================================================================
ru/engines/database-engines/sqlite.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/sqlite.html --> <nil>
ru/engines/database-engines/materialized-postgresql.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/materialized-postgresql.html --> <nil>
ru/engines/database-engines/replicated.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/replicated.html --> <nil>
ru/engines/database-engines/atomic.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/atomic.html --> <nil>
ru/engines/database-engines/lazy.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/lazy.html --> <nil>
ru/engines/database-engines/mysql.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/mysql.html --> <nil>
ru/engines/database-engines/postgresql.html
  bad reference: "parse \"/docs/category/инструкция-для-\\x00разработчиков\": net/url: invalid control character in URL" --- ru/engines/database-engines/postgresql.html --> <nil>
en/sql-reference/aggregate-functions/reference/stochasticlogisticregression.html
  hash does not exist --- en/sql-reference/aggregate-functions/reference/stochasticlogisticregression.html --> #stochasticlinearregression-usage-fitting
zh/sql-reference/aggregate-functions/reference/stochasticlogisticregression.html
  hash does not exist --- zh/sql-reference/aggregate-functions/reference/stochasticlogisticregression.html --> #stochasticlinearregression-usage-fitting
zh/engines/table-engines/integrations/hive.html
  hash does not exist --- zh/engines/table-engines/integrations/hive.html --> #在-clickhouse-中建表-2
========================================================================
✘✘✘ failed in 3.865139612s
10 errors in 2663 documents
INFO:root:Run failed
Run action done for: [Docs check]
```

<!---
A technical comment, you are free to remove or leave it as it is when PR is created
The following categories are used in the next scripts, update them accordingly
utils/changelog/changelog.py
tests/ci/cancel_and_rerun_workflow_lambda/app.py
-->
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

cc: @Felixoid @gingerwizard

DamianMaslanka5 committed:
Set docs check job status to fail when htmltest fails

Felixoid commented on docs/en/sql-reference/aggregate-functions/reference/stochasticlinearregression.md:
```diff
@@ -72,5 +72,5 @@ The query will return a column of predicted values. Note that first argument of
 **See Also**
 
-- [stochasticLogisticRegression](../../../sql-reference/aggregate-functions/reference/stochasticlogisticregression.md#stochasticlogisticregression)
+- [stochasticLogisticRegression](../../../sql-reference/aggregate-functions/reference/stochasticlogisticregression.md)
```
I am not sure about the relative path here and in other places

DamianMaslanka5 replied:
It works correctly, but could probably be simplified.

I removed the commit with markdown changes, because it got fixed in https://github.com/ClickHouse/ClickHouse/pull/74650

DamianMaslanka5 committed:
Ignore all zh and ru files

Links in those files sometimes contain null char (this looks like an issue with docusaurus/react) and some links are not working correctly

Felixoid commented:
@gingerwizard, please take a look at the HTML test. Should we ignore all directories completely?

Blargian commented:
> @gingerwizard, please take a look at the HTML test. Should we ignore all directories completely?

We're planning to generate Ru and Zh translations in the near future because they are completely outdated at the moment. It's okay to ignore Ru and Zh.

gingerwizard commented:
We're using docusaurus 3.7 now, id need to dig but this might a react 18.2 - we can upgrade if needed.  I'm +1 on this change
